tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -3143,14 +3155,43 @@ Status ReplaceSoftplusTanhAndMulWithMish(
   return OkStatus();
 }
 
+bool RuntimeFusionEnabled(const Cluster& cluster) {
+  static bool is_enabled = [&] {
+#if CUDNN_VERSION >= 8400
+    auto devices = cluster.GetDevices();
+    int num_gpus = 0;
+    int num_ampere = 0;
+    for (const auto& d : devices) {
+      if (d.second.type() == "GPU") {
+        num_gpus++;
+        auto cc_it = d.second.environment().find("architecture");
+        if (cc_it != d.second.environment().end()) {
+          double compute_capability = 0.0;
+          if (absl::SimpleAtod(cc_it->second, &compute_capability) &&
+              compute_capability >= 8.0) {
+            num_ampere++;
+          }
+        }
+      }
+    }
+    return CudnnUseFrontend() && CudnnUseRuntime() && num_gpus > 0 &&
+           num_gpus == num_ampere;
```

Comment:
This is not an Ampere only feature. It is actually supported on Volta and Turing. But we recommend to use the runtime fusion on Ampere or later GPUs mainly because the NVRTC might be very slow (esp. on Volta) and we haven't extensively tested the feature on older GPUs (like the heuristics). So, to be more precautious, I set this limitation.

I confirmed with our cudnn team if you (Waymo) think sm75 is important they will plan the work to test and create heuristics for it. How do you think?

Reply:
> because the NVRTC might be very slow (esp. on Volta) and we haven't extensively tested the feature on older GPUs (like the heuristics).

To be clear, you mean that the generated code might be slow, or that the compilation step might be slow?

> I confirmed with our cudnn team if you (Waymo) think sm75 is important they will plan the work to test and create heuristics for it. How do you think?

Yes, Waymo is on sm75.  AIUI we've discussed this feature up in meetings with our nvidia contacts for a long time now.  It seems like some wires got crossed and folks on the cudnn team didn't make it work well for hardware, while at the same time we were told that they were delivering this feature for us to use.

Anyway, you'e awesome, @kaixih, to be clear, I have no frustration towards you personally.

Reply:
I mean the complication step might be slow.

For the sm75 support, let me check with our cudnn team for details about what we already have and what is missing. I did all the development and experiments on Ampere but I am willing to add sm75 support if I can collect more info on this.

Reply:
> I mean the compilation step might be slow.

Why would compiling for sm75 be slower than compiling for sm80?

> For the sm75 support, let me check with our cudnn team for details about what we already have and what is missing. I did all the development and experiments on Ampere but I am willing to add sm75 support if I can collect more info on this.

Thanks a lot, Kaixi.

Reply:
> Why would compiling for sm75 be slower than compiling for sm80?

Our cudnn team told me that one of the major reason for slow compilation is the register spilling. If this happens, NVRTC needs time to figure it out. And I guess it is more likely to hit this on sm75 than sm80 in the current implementation. In fact, I found some cases that sm75 has much slower compilation time than sm80 and I'd already filed a bug to cudnn to work on.